database/sqltypes/sqltypes.go

```diff
@@ -239,7 +253,34 @@ func BuildValue(goval interface{}) (v Value, err error) {
 	}
 	return v, nil
 }
+// ConverAssignRowNullable is the same as ConvertAssignRow except that it allows
+// nil as a value for the row or any of the row values. In thoses cases, the
+// corresponding values are ignored.
+func ConvertAssignRowNullable(row []Value, dest ...interface{}) error {
+	if len(row) != len(dest) {
+		return errors.Newf(
+			"# of row entries %d does not match # of destinations %d",
+			len(row),
+			len(dest))
+	}
 
+	if row == nil {
+		return nil
+	}
+
+	for i := 0; i < len(row); i++ {
```

Comment:
I just copied the code below ConvertAssignRow and kept it as is for consistency. Also, here, you're actually iterating over dest and row, so you need the i